pkg/controller/volume/attachdetach/reconciler/reconciler_test.go

```diff
@@ -641,6 +641,80 @@ func Test_ReportMultiAttachError(t *testing.T) {
 	}
 }
 
+func Test_AddBackToReportAsAttachedList(t *testing.T) {
+	// Arrange
+	// create plugin for test detach failed
+	// (re-write detach function : fail to detach and return an detach error)
+	volumePluginMgr, fakePluginForDetachFailed := volumetesting.GetTestVolumePluginMgrForDetachFailed(t)
+	dsw := cache.NewDesiredStateOfWorld(volumePluginMgr)
+	asw := cache.NewActualStateOfWorld(volumePluginMgr)
+	fakeKubeClient := controllervolumetesting.CreateTestClient()
+	fakeRecorder := &record.FakeRecorder{}
+	fakeHandler := volumetesting.NewBlockVolumePathHandler()
+	ad := operationexecutor.NewOperationExecutor(operationexecutor.NewOperationGenerator(
+		fakeKubeClient,
+		volumePluginMgr,
+		fakeRecorder,
+		false, /* checkNodeCapabilitiesBeforeMount */
+		fakeHandler))
+	nsu := statusupdater.NewFakeNodeStatusUpdater(false /* returnError */)
+	reconciler := NewReconciler(
+		reconcilerLoopPeriod, maxWaitForUnmountDuration, syncLoopPeriod, false, dsw, asw, ad, nsu, fakeRecorder)
+
+	podName := "pod-uid"
+	volumeName := v1.UniqueVolumeName("volume-name")
+	volumeSpec := controllervolumetesting.GetTestVolumeSpec(string(volumeName), volumeName)
+	nodeName := k8stypes.NodeName("node-name")
+	dsw.AddNode(nodeName, false /*keepTerminatedPodVolumes*/)
+	volumeExists := dsw.VolumeExists(volumeName, nodeName)
+	if volumeExists {
+		t.Fatalf(
+			"Volume %q/node %q should not exist, but it does.",
+			volumeName,
+			nodeName)
+	}
+
+	generatedVolumeName, podErr := dsw.AddPod(types.UniquePodName(podName), controllervolumetesting.NewPod(podName, podName), volumeSpec, nodeName)
+	if podErr != nil {
+		t.Fatalf("AddPod failed. Expected: <no error> Actual: <%v>", podErr)
+	}
+
+	// Act
+	ch := make(chan struct{})
+	go reconciler.Run(ch)
+	defer close(ch)
+
+	fakePlugin := fakePluginForDetachFailed.FakeVolumePlugin
+	// Assert
+	waitForNewAttacherCallCount(t, 1 /* expectedCallCount */, fakePlugin)
+	// we expect 1 volume is in UpdateStatusFor cache
+	verifyNodesToUpdateStatusFor(t, 1 /* expectedCallCount */, nodeName, asw)
+
+	nodesForVolume := asw.GetNodesForVolume(generatedVolumeName)
+
+	// Act
+	podToDelete := podName
+	dsw.DeletePod(types.UniquePodName(podToDelete), generatedVolumeName, nodesForVolume[0])
+	//volumeExists := dsw.VolumeExists(generatedVolumeName, nodesForVolume[0])
+	if volumeExists {
+		t.Fatalf(
+			"Deleted pod %q from volume %q/node %q. Volume should also be deleted but it still exists.",
+			podToDelete,
+			generatedVolumeName,
+			nodesForVolume[0])
+	}
+
+	// sleep 20s, detach fail some times and reconciler will get exponential Back off error
```

Comment:
how long it takes to get exponential back off? I think 20s is little bit long?

Reply:
exponential back off follows the following sequence "0s 0.5s 1s 2s 4s ..."
I think 5s is enough